numpy/lib/_arraypad_impl.py

```diff
@@ -320,11 +323,20 @@ def _set_reflect_both(padded, axis, width_pair, method, include_edge=False):
     """
     left_pad, right_pad = width_pair
     old_length = padded.shape[axis] - right_pad - left_pad
-
+    
     if include_edge:
+        # Avoid wrapping with only a subset of the original area 
+        # by ensuring period can only be a multiple of the original 
+        # area's length.
+        old_length = old_length // original_period * original_period
         # Edge is included, we need to offset the pad amount by 1
         edge_offset = 1
     else:
+        # Avoid wrapping with only a subset of the original area 
+        # by ensuring period can only be a multiple of the original 
+        # area's length.
+        old_length = (old_length - 1) // (original_period - 1) \
+        * (original_period - 1) + 1
```

Comment:
Looks like the `+ 1` is subtracted out in the second line below. This line should also be indented for readability.

Reply:
Instead of line continuation, you can put the whole right hand side in parenthesis.